orchestration/dagster_orchestration/hca_orchestration/solids/load_hca/load_data_files.py

```diff
@@ -1,12 +1,230 @@
-from dagster import solid, Nothing, InputDefinition
+from typing import Iterator
+
+from dagster import composite_solid, solid, Nothing, InputDefinition, Int, configured, Failure
 from dagster.core.execution.context.compute import AbstractComputeExecutionContext
+from dagster.experimental import DynamicOutput, DynamicOutputDefinition
+from data_repo_client import ApiException
+from data_repo_client import JobModel
+from google.cloud import bigquery
+from google.cloud.bigquery.client import RowIterator
+
+from hca_manage.manage import JobId
+from hca_orchestration.contrib.bigquery import build_query_job_using_external_schema, build_extract_job
+from hca_orchestration.contrib.retry import is_truthy, retry
+from hca_orchestration.resources.config.hca_dataset import TargetHcaDataset
+from hca_orchestration.resources.config.scratch import ScratchConfig
+from hca_orchestration.solids.data_repo import wait_for_job_completion
+from hca_orchestration.support.typing import HcaScratchDatasetName
+from dagster_utils.typing import DagsterConfigDict
+from dagster_utils.contrib.google import gs_path_from_bucket_prefix
+
+
+FILE_LOAD_TABLE_BQ_SCHEMA = [
+    {
+        'mode': 'NULLABLE',
+        'name': 'source_path',
+        'type': 'STRING'
+    },
+    {
+        'mode': 'NULLABLE',
+        'name': 'target_path',
+        'type': 'STRING',
+    }
+]
+
+
+@solid(
+    required_resource_keys={"bigquery_client", "scratch_config", "storage_client", "target_hca_dataset"},
+    input_defs=[InputDefinition("scratch_dataset_name", HcaScratchDatasetName)],
+    output_defs=[DynamicOutputDefinition(name="control_file_path", dagster_type=str)]
+)
+def diff_file_loads(context: AbstractComputeExecutionContext,
+                    scratch_dataset_name: HcaScratchDatasetName) -> Iterator[str]:
+    """
+    Determine files to load by joining against the target HCA dataset, then drop the result in our scratch bucket as a
+     "control file" that Jade will use for bulk file ingest
+    :param context: Dagster solid context
+    :param scratch_dataset_name: Name of the "scratch" dataset we are using for this pipeline run
+    :return: Yields a list of control file blob names
+    """
+    target_hca_dataset = context.resources.target_hca_dataset
+    file_load_table_name = 'file_load_requests'
+    bigquery_client = context.resources.bigquery_client
+    scratch = context.resources.scratch_config
+
+    _determine_files_to_load(
```

Comment:
if we're just tossing away the return value of this job, where is the "determination" being tracked? in the scratch table?

i think we may want a comment to this effect.